pkg/devfile/generator/generators.go

```diff
@@ -55,54 +55,50 @@ func GetObjectMeta(name, namespace string, labels, annotations map[string]string
 	return objectMeta
 }
 
-// GetContainers iterates through the devfile components and returns a slice of the corresponding containers
+// GetContainers iterates through all container components, filters out init containers and returns corresponding containers
 func GetContainers(devfileObj parser.DevfileObj, options common.DevfileOptions) ([]corev1.Container, error) {
-	var containers []corev1.Container
-
-	options.ComponentOptions = common.ComponentOptions{
-		ComponentType: v1.ContainerComponentType,
-	}
-	containerComponents, err := devfileObj.Data.GetComponents(options)
+	allContainers, err := getAllContainers(devfileObj, options)
 	if err != nil {
 		return nil, err
 	}
-	for _, comp := range containerComponents {
-		envVars := convertEnvs(comp.Container.Env)
-		resourceReqs := getResourceReqs(comp)
-		ports := convertPorts(comp.Container.Endpoints)
-		containerParams := containerParams{
-			Name:         comp.Name,
-			Image:        comp.Container.Image,
-			IsPrivileged: false,
-			Command:      comp.Container.Command,
-			Args:         comp.Container.Args,
-			EnvVars:      envVars,
-			ResourceReqs: resourceReqs,
-			Ports:        ports,
+
+	// filter out init containers
+	preStartEvents := devfileObj.Data.GetEvents().PreStart
+	if len(preStartEvents) > 0 {
+		var eventCommands []string
+		commands, err := devfileObj.Data.GetCommands(common.DevfileOptions{})
+		if err != nil {
+			return nil, err
 		}
-		container := getContainer(containerParams)
 
-		// If `mountSources: true` was set PROJECTS_ROOT & PROJECT_SOURCE env
-		if comp.Container.MountSources == nil || *comp.Container.MountSources {
-			syncRootFolder := addSyncRootFolder(container, comp.Container.SourceMapping)
+		commandsMap := common.GetCommandsMap(commands)
 
-			projects, err := devfileObj.Data.GetProjects(common.DevfileOptions{})
-			if err != nil {
-				return nil, err
-			}
-			err = addSyncFolder(container, syncRootFolder, projects)
-			if err != nil {
-				return nil, err
+		for _, event := range preStartEvents {
+			eventSubCommands := common.GetCommandsFromEvent(commandsMap, event)
+			eventCommands = append(eventCommands, eventSubCommands...)
+		}
+
+		for _, commandName := range eventCommands {
+			if command, ok := commandsMap[commandName]; ok {
```

Comment:
this check is really not needed, its guaranteed to be in commands Map via `GetCommandsFromEvent()`

Reply:
you are right. updated